languages.toml

```diff
@@ -335,39 +391,51 @@ file-types = ["sh", "bash", "zsh", ".bash_login", ".bash_logout", ".bash_profile
 shebangs = ["sh", "bash", "dash"]
 roots = []
 comment-token = "#"
-
 language-server = { command = "bash-language-server", args = ["start"] }
 indent = { tab-width = 2, unit = "  " }
 
+[[grammar]]
+name = "bash"
+source = { git = "https://github.com/tree-sitter/tree-sitter-bash", rev = "a8eb5cb57c66f74c63ab950de081207cccf52017" }
+
 [[language]]
 name = "php"
 scope = "source.php"
 injection-regex = "php"
 file-types = ["php"]
 shebangs = ["php"]
 roots = []
-
 indent = { tab-width = 4, unit = "    " }
 
+[[grammar]]
+name = "php"
+source = { git = "https://github.com/tree-sitter/tree-sitter-php", rev = "57f855461aeeca73bd4218754fb26b5ac143f98f" }
+
```

Comment:
Any reason for keeping this separate? I'd make `source` a part of the `[[language]]` table (if no source specified, we just skip it in the fetcher/builder)

Reply:
I separated them out to make it easier for languages to share grammars, but it occurs to me that we can still do that if the sources live under `[[language]]` because of the `grammar` key (changed from `tree_sitter_library`)

I'll see if I can put them back under language :+1:

Reply:
I think we can just have the grammar configuration as a table under `LanguageConfiguration`, so

```toml
[[language]]
name = "erlang"
grammar.source = { git = "..", rev = ".." }
```

in the simple case and when a grammar needs extended configuration it can be done like

```toml
[[language]]
name = "tsx"

[language.grammar]
source = { git = "..", rev = ".." }
path = "tsx"
compiler-standard = "c99"
```

Reply:
llvm-mir-yaml uses the yaml grammar so I'm imagining that will be configured like so:

```toml
[[language]]
name = "llvm-mir-yaml"
# no 'grammar' key
tree-sitter-library = "yaml"
```

so it can use the compiled yaml grammar without having to specify the source/path/compiler-opts

Reply:
But then aren't we depending on 1. the YAML grammar to be in use and 2. that it is downloaded and compiled first?

Reply:
Hmm true, that is handled better with a separate grammars array :thinking: 

I'm kind of split on this now, it's a little cleaner to have everything under `[[languages]]` but less flexible when it comes to re-using grammars